### PR TITLE
fix(ConnectableObservable): fix ConnectableObservable connection handling issue

### PR DESCRIPTION
**Pull request by trxcllnt**
**Description:**
Fixes an issue that caused ConnectableObservable to assign itself a connection that should be disposed when working with synchronous Observables. This issue caused a race condition in synchronous refCounted Observables if the subscribers dispose before the published source completes, causing the ConnectableObservable not to re-subscribe to the original source when it should have.

**Related issue (if exists):**
#1718

### CODE REVIEW

File: src/observable/ConnectableObservable.ts [hunk 1]
```diff
@@ -123,9 +129,11 @@ class RefCountSubscriber<T> extends Subscriber<T> {
     }
 
     const { connection } = this;
-    if (connection) {
-      this.connection = null;
-      connection.unsubscribe();
+    const sharedConnection = (<any> connectable)._connection;
+    this.connection = null;
+
+    if (sharedConnection && (!connection || sharedConnection === connection)) {
```
Comment (blesh) [inline src/observable/ConnectableObservable.ts]: Can you add a comment explaining this line? I don't want to grok it again later. haha.

**Comment by blesh** [comment]
Other than the one comment I made, LGTM.

File: src/observable/ConnectableObservable.ts (hunk 1 above)
Comment (trxcllnt) [inline src/observable/ConnectableObservable.ts]: Sure can.

**Comment by blesh** [comment]
:shipit:

**Pull request merged**